Endava.TechCourse.BankApp.Server/Controllers/WalletsController.cs

```diff
@@ -19,41 +19,48 @@ public WalletsController(ApplicationDbContext context)
     }
 
     [HttpPost]
-    public async Task<IActionResult> CreateWallet([FromBody] WalletDto walletDto)
+    public async Task CreateWallet([FromBody] WalletDto walletDto)
     {
-        var currency = await this.context.Currencies.FirstOrDefaultAsync(x => x.Id.ToString() == walletDto.Currency.Id);
+        // searching by 'x.Id.ToString()'
+        // this works in prod, but fails in UnitTests (FirstOrDefault/Async also fails in UnitTests)
+        // var currency = await context.Currencies.FirstAsync(x => x.Id.ToString() == walletDto.Currency.Id);
+
+        // searching by 'x.Id' (after converting the request string to Guid)
+        // this works in prod, also in UnitTests
+        if (!Guid.TryParse(walletDto.Currency.Id, out var requestCurrencyId))
+            throw new InvalidOperationException($"Received invalid Id: {walletDto.Currency.Id}");

```

Comment:
This is a controller. If your data are not valid , you can return a BadRequest(errorMessage)